### ISSUE TITLE
loss 'nan' && error rate hasn't  been changed

### ISSUE THREAD

**Issue by namedBen** [AcrossV/Gated-XNOR#1]
when i run your code file 'mnist_GXNOR.py', the print-log shows like:
______________________________________
Epoch 2 of 4000 took 12.7568740845s
  update_type:                   200
  LR:                            0.0996552080135
  th:                            3.0
  LR_decay:                      0.996552080135
  training loss:                 nan
  validation loss:               nan
  validation error rate:         90.08413461538461%
  best epoch:                    3
  best test error rate:          90.1943108974359%
  test loss:                     nan
  test error rate:               90.1943108974359%
______________________________________
Epoch 400 of 4000 took 12.7652070522s
  update_type:                   200
  LR:                            0.0993116048421
  th:                            3.0
  LR_decay:                      0.996552080135
  training loss:                 nan
  validation loss:               nan
  validation error rate:         90.08413461538461%
  best epoch:                    400
  best test error rate:          90.1943108974359%
  test loss:                     nan
  test error rate:               90.1943108974359%
______________________________________
it seems that loss or error-rate hasn't  been changed in every epoch while training.The same problem also exits when running "cifar10_GXNOR.py"

**Comment by AcrossV** [comment]
I think the reason is not the code. Maybe you can try other simple codes and look if the problem occurs. I Tried our code and the problem you said did not occur.

**Comment by namedBen** [comment]
thanks, i have already solved the problem
